### PR TITLE
Node key manager refactoring

### PR DESCRIPTION
**Pull request by MarekM25**

### CODE REVIEW

File: src/Nethermind/Nethermind.KeyStore/PasswordProvider.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                    var character = (char)stream.Read();
+                    if (char.IsWhiteSpace(character))
+                    {
+                        whitespaces.Add(character);
+                    }
+                    else
+                    {
+                        if (whitespaces.Count != 0)
+                        {
+                            FillWhitespaceList(secureString, whitespaces);
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.KeyStore/PasswordProvider.cs:87]: We Trim() file content before. When we change to reading from the stream, I have to introduce that logic.

**Review by tkstanczak — APPROVED** [review]

File: src/Nethermind/Nethermind.KeyStore/PasswordProvider.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            {
+                passwordFromFile = GetPassword(index);
+            }
+
+            return passwordFromFile != null ? passwordFromFile : GetPasswordFromConsole();
+        }
+
+        public SecureString GetPassword(int keyStoreConfigPasswordIndex)
+        {
+            string GetPasswordN(int n, string[] passwordsCollection) => passwordsCollection?.Length > 0 ? passwordsCollection[Math.Min(n, passwordsCollection.Length - 1)] : null;
```
[the remark this answers is not in the document]
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.KeyStore/PasswordProvider.cs:49]: Passwords is not recommended and more for debug purposes. They shouldn't be used together, so you should have either one or the other.

File: src/Nethermind/Nethermind.KeyStore/PasswordProvider.cs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+            SecureString password = null;
+            var passwordFile = GetPasswordN(keyStoreConfigPasswordIndex, _keyStoreConfig.PasswordFiles);
+            if (passwordFile != null)
+            {
+                string blockAuthorPasswordFilePath = passwordFile.GetApplicationResourcePath();
+                password = File.Exists(blockAuthorPasswordFilePath)
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.KeyStore/PasswordProvider.cs:56]: in original it was  ove IFileSystem

File: src/Nethermind/Nethermind.KeyStore/PasswordProvider.cs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+                    if (char.IsWhiteSpace(character))
+                    {
+                        whitespaces.Add(character);
+                    }
+                    else
+                    {
+                        if (whitespaces.Count != 0)
+                        {
+                            FillWhitespaceList(secureString, whitespaces);
+                            whitespaces = new List<char>();
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.KeyStore/PasswordProvider.cs:88]: just whitespaces.Clear() ?

File: src/Nethermind/Nethermind.KeyStore/PasswordProvider.cs (hunk 1 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.KeyStore/PasswordProvider.cs:87]: Trim removes also from begining

File: src/Nethermind/Nethermind.KeyStore/PasswordProvider.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (MarekM25) [reply src/Nethermind/Nethermind.KeyStore/PasswordProvider.cs:49]: > Passwords is not recommended and more for debug purposes. They shouldn't be used together, so you should have either one or the other.

That explains lots of things. Thanks!

File: src/Nethermind/Nethermind.KeyStore/PasswordProvider.cs (hunk 3 above)
Comment (MarekM25) [reply src/Nethermind/Nethermind.KeyStore/PasswordProvider.cs:56]: I know. However, it was used for mocking in tests. Now, we did it in another way, and we don't need it in tests anymore. Thanks to that, we reduced the number of injected dependencies. If you think that we should return it, I will do it.

File: src/Nethermind/Nethermind.KeyStore/PasswordProvider.cs (hunk 4 above)
Comment (MarekM25) [reply src/Nethermind/Nethermind.KeyStore/PasswordProvider.cs:88]: > just whitespaces.Clear() ?

Argh... thx ! :)

**Pull request merged**